File: server/src/main/java/message/csmessage/usermanagement/CsRegister.java
package message.csmessage.usermanagement;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import message.csmessage.CsMessage;
import session.Session;

@AllArgsConstructor
public class CsRegister extends CsMessage {
    public String userName;
    public String password;
    public String retypePassword;

    @JsonCreator
    public CsRegister(@JsonProperty("generatedTime") long generatedTime,
                      @JsonProperty("userName") String userName,
                      @JsonProperty("password") String password,
                      @JsonProperty("retypePassword") String retypePassword) {
        this.generatedTime = generatedTime;
        this.userName = userName;
        this.password = password;
        this.retypePassword = retypePassword;
    }

    @Override
    public void onMessage(Session session) {
        session.onRegister(userName, password, retypePassword);
    }
}

File: Client/Fruitster/Assets/Script/NetConnection/Messages/CsMessages/UserValidation/CsRegister.cs
using System;

namespace Script.Messages.CsMessages
{
    [Serializable]
    public class CsRegister : CsMessage
    {
        public string userName;
        public string password;
        public string retypePassword;

        public CsRegister(string userName, string password, string retypePassword)
        {
            this.userName = userName;
            this.password = password;
            this.retypePassword = retypePassword;
            
            UserProperties.MainPlayer.userName = userName;
        }
    }
}